src/viewport.js

```diff
@@ -734,6 +734,33 @@ $.Viewport.prototype = {
 
 
     /**
+     * Returns bounds taking constrains into account
+     * Added to improve constrained panning
+     * @param {Boolean} immediately
+     * @return {OpenSeadragon.Viewport} Chainable.
+     */
+    // Added to improve constrained panning
+    getConstrainedBounds: function( immediately ) {
+        var actualZoom = this.getZoom(),
+            constrainedZoom = Math.max(
+                Math.min( actualZoom, this.getMaxZoom() ),
+                this.getMinZoom()
+            ),
+            bounds,
+            constrainedBounds;
+
+        if ( actualZoom != constrainedZoom ) {
+            this.zoomTo( constrainedZoom, this.zoomPoint, immediately );
```

Comment:
I am a bit worried that we are calling `this.zoomTo` and `this._applyBoundaryConstraints` in a `getXXX` method.
It seems like it is doing more than what the name suggest.